Use emaint for syncing and for applying package moves

kerframil opened the pull request:
By using emaint for syncing, EMERGE_DEFAULT_OPTS is ignored and, thus, it
will not break if a user has --ask as a default option (as some do). Note that
emerge --sync calls through to the emaint sync system anyway.

Also, emaint provides a specific option for applying package moves so use it in place of emerge --info.

Commit:
Use emaint for syncing and for applying package moves

By using emaint for syncing, EMERGE_DEFAULT_OPTS is ignored and, thus, it
will not break if a user has --ask as a default option (as some do). Note that
emerge --sync calls through to the emaint sync system anyway.

Also, emaint provides a specific option for applying package moves so use it.

GreatEmerald commented:
emaint also syncs layman repos now, so in addition to this the layman syncing step should be disabled.

Commit:
Merge branch 'master' of https://github.com/gentoo/porticron

Commit:
Revise README.rst and porticron.conf and drop SYNC_OVERLAYS_CMD

The previous commit merged upstream changes while also removing support
for SYNC_OVERLAYS_CMD. In this day and age, portage supports a formal
repo configuration syntax and there is really no need to specify a
distinct command. Instead, default to emaint sync -a, which
automatically syncs all repos that have "auto-sync = yes" in effect. To
account for this change, also provide a commented sample SYNC_CMD that
syncs only the gentoo repo - thus, avoiding any overlays.

Commit:
Add mock test for emaint(1)

Commit:
Update expected porticron output in lieu of emaint usage

Commit:
Update test.bats to use the mock emaint

ghost commented:
Pull request updated to address the comment from @GreatEmerald, with test support added and passed. Please review.

Commit:
Eliminate accidental UK English spelling in favour of US English

hartwork commented:
Hi! Will need to come back later for a closer look. Could you get the merge commit out of the commit list? That would be cool. Thanks!

ghost commented:
I don't know how to do that but I'll see if I can figure it out. Otherwise, I'll do it all over and overwrite my branch.
